src/derivkit/local_polynomial_derivative/diagnostics.py

```diff
@@ -0,0 +1,86 @@
+"""Diagnostics for local polynomial derivative estimation."""
+
+from __future__ import annotations
+
+from typing import Any, Dict
+
+import numpy as np
+
+from .fit import design_matrix
+from .local_poly_config import LocalPolyConfig
+
+__all = ["make_diag"]
+
+
+def make_diag(
+    x0: float,
+    config: LocalPolyConfig,
+    xs: np.ndarray,
+    ys: np.ndarray,
+    keep: np.ndarray,
+    coeffs: np.ndarray,
+    degree: int,
+    order: int,
+    ok: bool,
+) -> Dict[str, Any]:
+    """Builds diagnostics dictionary.
+
+    Args:
+        x0:
+            The center point for polynomial fitting.
+        config:
+            LocalPolyConfig instance with fitting settings.
+        xs:
+            An array of sample points (shape (n_samples,)).
+        ys:
+            An array of function evaluations (shape (n_samples, n_components)).
+        keep:
+            A boolean array indicating which samples were used (shape (n_samples,)).
```

Comment:
```suggestion
        keep:
            A boolean array indicating which samples were used (shape (n_samples,)).
```
I'm not sure if `keep` and "samples" are the right words here; it's the data points that were used in the fit. Maybe `used` instead of `keep` and "grid points" instead of "samples"?

Reply:
I tihn ksample points can remain hmm because it is sampling in a way. WDYT? i think it is more appopriate

Reply:
"sample" to me carries a connotation of randomly chosing a subset of something. This isn't really the case here, right?

Reply:
kinda but i still tihnk this is a valid choice (of word). it is sampling on a grid so sohuld be fine. sampling is not just mcmc and nested sampling and stuff liek that